File: vurtual/abc147_e.py
# https://atcoder.jp/contests/abc147/tasks/abc147_e
import sys
# input = sys.stdin.buffer.readline
def input(): return sys.stdin.readline().rstrip()
sys.setrecursionlimit(10 ** 7)

H, W = map(int, input().split())
A = [[int(x) for x in input().split()] for _ in range(H)]
B = [[int(x) for x in input().split()] for _ in range(H)]
C = [[0] * W for _ in range(H)]
for i in range(H):
    for j in range(W):
        C[i][j] = abs(A[i][j] - B[i][j])
center = 80 * (H+W) # (80+80)*80=12800

dp = [[0] *W for _ in range(H)]
dp[0][0] = (1 << (center - C[0][0])) | (1 << (center + C[0][0]))
for i in range(H):
    for j in range(W):
        if i:
            dp[i][j] |= dp[i-1][j] << C[i][j]
            dp[i][j] |= dp[i-1][j] >> C[i][j]
        if j:
            dp[i][j] |= dp[i][j-1] << C[i][j]
            dp[i][j] |= dp[i][j-1] >> C[i][j]
ans = center * 2
for i in range(center*2):
    if (dp[H-1][W-1] >> i) & 1:
        ans = min(ans, abs(i-center))
print(ans)


# TLE
# H, W = map(int, input().split())
# A = [[int(x) for x in input().split()] for _ in range(H)]
# B = [[int(x) for x in input().split()] for _ in range(H)]
# C = [[0] * W for _ in range(H)]
# for i in range(H):
#     for j in range(W):
#         C[i][j] = abs(A[i][j] - B[i][j])

# dp = [[set() for _ in range(W)] for _ in range(H)]
# dp[0][0].add(C[0][0])
# for i in range(H):
#     for j in range(W):
#         if i:
#             for k in dp[i-1][j]:
#                 dp[i][j].add(abs(k+C[i][j]))
#                 dp[i][j].add(abs(k-C[i][j]))
#         if j:
#             for k in dp[i][j-1]:
#                 dp[i][j].add(abs(k+C[i][j]))
#                 dp[i][j].add(abs(k-C[i][j]))
# print(min(dp[H-1][W-1]))

File: abc147_e.cpp
// https://atcoder.jp/contests/abc147/tasks/abc147_e
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define all(v) (v).begin(),(v).end()
#define chmin(x,y) x = min(x,y)
#define chmax(x,y) x = max(x,y)
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

const int D = 80*160+10;    // -80*160:0, 0:D, 80*160:2D
const int D2 = D*2;

int main() {
	int H, W;
	cin >> H >> W;
    vector<vector<int>> A(H, vector<int>(W));
    rep(i, H) rep(j, W) {
        int x;
        cin >> x;
        A[i][j] =x;
    }
    rep(i, H) rep(j, W) {
        int x;
        cin >> x;
        A[i][j] = abs(x - A[i][j]);
    }
    vector<vector<bitset<D2>>> dp(H, vector<bitset<D2>>(W));
    dp[0][0][D-A[0][0]] = 1;
    dp[0][0][D+A[0][0]] = 1;
    rep(i, H) rep(j, W) {
        if (i) {
            dp[i][j] |= dp[i-1][j] << A[i][j];
            dp[i][j] |= dp[i-1][j] >> A[i][j];
        }
        if (j) {
            dp[i][j] |= dp[i][j-1] << A[i][j];
            dp[i][j] |= dp[i][j-1] >> A[i][j];
        }
    }

    int ans = D2;
    rep(i,D2) if (dp[H-1][W-1][i]) {
        ans = min(ans, abs(i-D));
    }
	cout << ans << endl;
	return 0;
}
